### ISSUE TITLE
allow markdown in changeset comments

### ISSUE THREAD

**Issue by fuddl** [openstreetmap/openstreetmap-website#844]
would be nice to link to something like another edit or a wiki page. or posting pictures

**Comment by planemad** [comment]
Would love to see this. Having images (even thumbnails) adds great context to conversation and makes the thread more readable.

**Comment by pnorman** [comment]
Do we do Markdown on anything API accessible?

**Comment by andrewharvey** [comment]
would be nice to use selected wikitext templates like `{{tag|informal|yes}}` or alternatively just parse out a pasted URL like `https://wiki.openstreetmap.org/wiki/Tag:informal%3Dyes` and display it as [`informal`](https://wiki.openstreetmap.org/wiki/Key:informal)=[`yes`](https://wiki.openstreetmap.org/wiki/Tag:informal%3Dyes).

Suggesting tags to use within a changeset comment means either
1. enter the tag as `key=value` without any links (but then especially new users don't know what that means or where to read up about it)
2. enter the full URL but this makes the comment long and hard to read the exactly key=value from the url.
3. do both of the above, but the comment is still too long and harder to read

**Comment by AntonKhorev** [comment]
> would be nice to use selected wikitext templates like {{tag|informal|yes}} or alternatively just parse out a pasted URL like https://wiki.openstreetmap.org/wiki/Tag:informal%3Dyes

But will those writing comments be bothered to paste the correct urls/templates?

If we just parse `key=value` (and maybe also `key=*`) as tag references, how many false positives/negatives we'll get?

**Comment by andrewharvey** [comment]
> But will those writing comments be bothered to paste the correct urls/templates?

Good point, then perhaps as you suggest parsing key=value would be okay and/or reducing a pasted full wiki link.

Although the more we do this automatically the more important a preview/edit button is.

**Comment by 1ec5** [comment]
Many websites that accept Markdown end up pretty-printing raw URLs. For example, GitHub turns `https://github.com/openstreetmap/openstreetmap-website/issues/844#issuecomment-2705479356` into https://github.com/openstreetmap/openstreetmap-website/issues/844#issuecomment-2705479356. It isn’t the most discoverable syntax: you’d probably see people occasionally trying to replicate the presentation by hand as ```[`informal=yes`](https://wiki.openstreetmap.org/wiki/Tag:informal%3Dyes)```, but I don’t think that would be a huge problem.

**Comment by andrewharvey** [comment]
> Many websites that accept Markdown end up pretty-printing raw URLs. For example, GitHub turns https://github.com/openstreetmap/openstreetmap-website/issues/844#issuecomment-2705479356 into https://github.com/openstreetmap/openstreetmap-website/issues/844#issuecomment-2705479356. It isn’t the most discoverable syntax: you’d probably see people occasionally trying to replicate the presentation by hand as [`informal=yes`](https://wiki.openstreetmap.org/wiki/Tag:informal%3Dyes), but I don’t think that would be a huge problem.

Good idea! I think this is getting too far from the ticket here which is about user entered markdown content to be formatted, so I've opened #5780 for what you describe.
